src/workload.rs

```diff
@@ -213,7 +217,16 @@ impl xds::Handler<XdsWorkload> for Arc<Mutex<WorkloadStore>> {
         let mut wli = self.lock().unwrap();
         let handle = |res: XdsUpdate<XdsWorkload>| {
             match res {
-                XdsUpdate::Update(w) => wli.insert_xds_workload(w.resource)?,
+                XdsUpdate::Update(w) => {
+                    // prefetch CA cert on discovery
+                    let identity = &Identity::Spiffe {
```

Comment:
my be better to do it in insert_xds_workload, then we have already converted XdsWorkload to Workload can can do `workload.identity()`. Notably, the logic here differs from `identity()`